server/util/networking/networking.go

```diff
@@ -210,6 +210,24 @@ func DeleteRuleIfSecondaryNetworkEnabled(ctx context.Context, vmIdx int) error {
 	return runCommand(ctx, "ip", "rule", "del", "from", getCloneIP(vmIdx))
 }
 
+func routeExists(ctx context.Context, source string, gateway string) (bool, error) {
+	b, err := sudoCommand(ctx, "ip", "route")
+	if err != nil {
+		return false, err
+	}
+	output := strings.TrimSpace(string(b))
+	if len(output) == 0 {
+		return false, nil
+	}
+	providedRoute := source + " via " + gateway
+	for _, route := range strings.Split(output, "\n") {
+		if strings.HasPrefix(route, providedRoute) {
```

Comment:
nit: can you add a comment with an example output line, so that it's more clear why we need `HasPrefix` vs `==`?